tests/sanity/sanity_test.go

```diff
@@ -0,0 +1,126 @@
+	d "github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	instanceID       = "i-1234567890abcdef0"
+	region           = "us-west-2"
+	availabilityZone = "us-west-2a"
+)
+
+var (
+	fakeMetadata = &metadata.Metadata{
+		InstanceID: instanceID,
+		Region:     region,
+	}
+	outpostArn = &arn.ARN{
+		Partition: "aws",
+		Service:   "outposts",
+		Region:    "us-west-2",
+		AccountID: "123456789012",
+		Resource:  "op-1234567890abcdef0",
+	}
+)
+
+func TestSanity(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Test panicked: %v", r)
+		}
+	}()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tmpDir, err := os.MkdirTemp("", "csi-sanity-")
+	if err != nil {
+		t.Fatalf("Failed to create sanity temp working dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	defer func() {
+		if err = os.RemoveAll(tmpDir); err != nil {
+			t.Fatalf("Failed to clean up sanity temp working dir %s: %v", tmpDir, err.Error())
+		}
+	}()
+
+	endpoint := fmt.Sprintf("unix:%s/csi.sock", tmpDir)
+	mountPath := path.Join(tmpDir, "mount")
+	stagePath := path.Join(tmpDir, "stage")
+
+	fakeMounter, FakeCloud, fakeMetadataService, fakeKubernetesClient := createMockObjects(mockCtrl, mountPath)
+
+	driverOptions := &d.Options{
+		Mode:                              d.AllMode,
+		ModifyVolumeRequestHandlerTimeout: 60,
+		Endpoint:                          endpoint,
+	}
+
+	drv, err := d.NewDriver(FakeCloud, driverOptions, fakeMounter, fakeMetadataService, fakeKubernetesClient)
+	if err != nil {
+		t.Fatalf("Failed to create fake driver: %v", err.Error())
+	}
+	go func() {
+		if err := drv.Run(); err != nil {
+			panic(fmt.Sprintf("%v", err))
+		}
+	}()
+
+	config := csisanity.TestConfig{
+		TargetPath:                  mountPath,
+		StagingPath:                 stagePath,
+		Address:                     endpoint,
+		DialOptions:                 []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())},
+		IDGen:                       csisanity.DefaultIDGenerator{},
+		TestVolumeSize:              10 * util.GiB,
+		TestVolumeAccessType:        "mount",
+		TestVolumeMutableParameters: map[string]string{"iops": "3014", "throughput": "153"},
+		TestVolumeParameters:        map[string]string{"type": "gp3", "iops": "3000"},
+	}
+	csisanity.Test(t, config)
+}
+
+func createMockObjects(mockCtrl *gomock.Controller, mountPath string) (*FakeMounter, *FakeCloud, *metadata.MockMetadataService, *d.MockKubernetesClient) {
+	fakeMounter := NewFakeMounter()
+	fakeCloud := newFakeCloud(*fakeMetadata, mountPath)
+	fakeKubernetesClient := d.NewMockKubernetesClient(mockCtrl)
+	fakeMetadataService := metadata.NewMockMetadataService(mockCtrl)
+
+	fakeMetadataService.EXPECT().GetAvailabilityZone().Return(availabilityZone).AnyTimes()
```

Comment:
Preference: I would fake MetadataService instead of mocking it, so I wouldn't have to hook these expects/constants up within this function (our sanity tests should not care about what AZ we're in), but will lgtm the pr. 

Thanks for making sure we catch any CSI spec regressions!

Reply:
+1. Any reason why we're faking `cloud` and `mounter` but not `metadata`?

Reply:
Updated to fake metadata in latest rev thanks for the feedback